### DISCUSSION TITLE
NotificationReceived event firing when app is not in foreground

### DISCUSSION THREAD

**Discussion by chuckgmthompson** [thomasgalliker/Plugin.FirebasePushNotifications#45] [General]
I have a handler for OnNotificationReceived and it works and fires great when the app is in the foreground.  but if the app is not in the foreground, even with the debugger attached in emulator, the event is not fired when a notification is received from FCM.  I know it in received because this pluggin receives it logs it and shows the notification.  Is there a way to have the handler fire even if the app is not in foreground?

**Comment by thomasgalliker** [comment]
Silly question: Did you ask for the notification permissions?
